backend/frontend/src/screens/BookingInfoScreen.js

```diff
@@ -339,6 +346,15 @@ function BookingInfoScreen() {
     }
   }, [isLogin, userLoginSuccess]);
 
+  
+  useEffect(() => {
+    const theCourt = courts?.find((court) => court.name === courtName);
+    const courtId = theCourt?.id;
+    if (slots?.length !== 0 && additionalSlots?.length !== 0 && courtId) {
+      dispatch(getNearestSlot(courtId, date));
+    }
+  }, [dispatch, courtName, slots, additionalSlots, courts, date]);
```

Comment:
NIT : Dependencies need to be in Alpabetical order.

Reply:
done